File: src/main/java/eu/qwsome/sql/condition/Between.java
package eu.qwsome.sql.condition;

/**
 * @author Lukáš Kvídera
 */
class Between implements Condition {

  private final ValueHolder field;
  private final ValueHolder from;
  private final ValueHolder to;


  Between(final ValueHolder field, final ValueHolder from, final ValueHolder to) {
    this.field = field;
    this.from = from;
    this.to = to;
  }

  @Override
  public CharSequence get() {
    final StringBuilder builder = new StringBuilder();
    appendTo(builder);
    return builder;
  }

  @Override
  public void appendTo(final StringBuilder builder) {
    builder.append(this.field.getSql())
      .append(" BETWEEN ")
      .append(this.from.getSql())
      .append(" AND ")
      .append(this.to.getSql());
  }

  @Override
  public ValueConstructor getValues() {
    return new ValueConstructor()
      .add(this.field)
      .add(this.from)
      .add(this.to);
  }
}

File: src/main/kotlin/eu/qwsome/sql/condition/Between.kt
/*
 * @author Lukáš Kvídera
 */
package eu.qwsome.sql.condition

class Between(
    private val field: ValueHolder,
    private val from: ValueHolder,
    private val to: ValueHolder,
) : Condition {

    override fun appendTo(builder: StringBuilder) {
        builder.append(field.getSql())
            .append(" BETWEEN ")
            .append(from.getSql())
            .append(" AND ")
            .append(to.getSql())
    }

    override fun getValues(): ValueConstructor {
        return ValueConstructor().addAll(field, from, to)
    }
}
